modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderItemLocalServiceImpl.java

```diff
@@ -2017,6 +2089,37 @@ private CommerceOrderItem _updateCommerceOrderItem(
 		return commerceOrderItemPersistence.update(commerceOrderItem);
 	}
 
+	private CommerceOrderItem _updateCommerceOrderItem(
+			long commerceOrderItemId, int quantity, String json,
+			ServiceContext serviceContext, boolean orderAdmin)
+		throws PortalException {
+
+		CommerceOrderItem commerceOrderItem =
+			commerceOrderItemPersistence.findByPrimaryKey(commerceOrderItemId);
+
+		CommerceOrder commerceOrder = commerceOrderItem.getCommerceOrder();
+
+		if (!orderAdmin) {
```

Comment:
when orderAdmin if false this method is exaclty the same as the old one, so why making a copy instead of updating that one? same for all of the methods added

Reply:
this way you are skipping all the validations, but we should exclude only those coming from _commerceOrderValidatorRegistry.validate()